lib/src/modules/compiler/template_parser/GRAMMAR.md

```diff
@@ -0,0 +1,38 @@
+#Template Grammar
+
+###Todos
+* Can you interpolate comments?
+* Can you have multiple pipes?
```

Comment:
Just replied - the answer is yes. This is one area where it _won't_ match the Dart expression AST, so we'll need the expression syntax to be something like `Expression (| Pipe)?`.